jadx-core/src/main/java/jadx/core/dex/visitors/blocks/BlockExceptionHandler.java

```diff
@@ -65,6 +65,10 @@ public static boolean process(MethodNode mth) {
 			removeMonitorExitFromExcHandler(mth, eh);
 		}
 		BlockProcessor.removeMarkedBlocks(mth);
+
+		List<BlockNode> sorted = new ArrayList<>(mth.getBasicBlocks().size());
+		BlockUtils.dfsVisit(mth, sorted::add);
+		removeUnusedExcHandlers(mth, tryBlocks, sorted);
```

Comment:
Looks like `sorted` blocks list later used only with `contains` so it is better to use `Set` instead `List`.
Or even better to use `BlockSet` from [here](https://github.com/skylot/jadx/blob/5aaceee978dba7935276672c26f0143e07072919/jadx-core/src/main/java/jadx/core/utils/blocks/BlockSet.java#L13)

Reply:
Done